### PR TITLE
Fixing Python3/unicode issues

### PR DESCRIPTION
**Pull request by martinm82**
This PR is an attempt to fix the Unicode decoding errors reported in https://github.com/distcc/distcc/issues/224.

### CODE REVIEW

File: include_server/test_data/supported_encoding.cpp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+�
```
Comment (TafThorne) [inline include_server/test_data/supported_encoding.cpp]: Is this change needed?  It might just be the web interface on GitHub making it look a little strange to me.

**Comment by TafThorne** [comment]
As @afbjorklund asked for the request, if he could green light the changes I am happy to do the merging.  Made my one query from a quick look over the code.  Again, not a Python guy so cannot comment much.

File: include_server/test_data/supported_encoding.cpp (hunk 1 above)
Comment (thinrope) [reply include_server/test_data/supported_encoding.cpp]: Probably... it is an "empty" file in ISO-8859 encoding (as reported by `file`).

Comment (martinm82) [reply include_server/test_data/supported_encoding.cpp]: Yes, it is actually an empty file with ISO-8859 encoding used by the unit test

**Comment by martinm82** [comment]
I ran also the changes against the full travis ci configuration I created in another PR in my personal branch: https://travis-ci.org/tt-gf/distcc/builds/275796576. This branch contains as well other changes but as a prove that the Python3 changes work also on the other configuration it should be enough.

**Comment by afbjorklund** [comment]
@TafThorne: Actually I thought the whole python3 thing a bit premature to start with, but since it has been merged to master it would be great if it could made functional again (i.e. fix the error reported in #224).

**Comment by timofonic** [comment]
Any news? Why is the distcc project stagnating while icecream gets developed faster? What about merging both projects and avoid the mess?

**Comment by afbjorklund** [comment]
@timofonic : I'm not sure what the main differences/benefits between icecream and distcc are ?
The source bases look quite different though, even different languages (C/C++) and everything.

**Comment by martinm82** [comment]
Basically the architecture is quite different in both projects, icecream uses a client/server approach with the icecc daemons and the schedulers that are dispatching the compile requests to the daemons. Also icecream is strictly depending on doing preprocessing locally and depending on toolchain environments (ICECC_VERSION).

Whereas distcc supports remote preprocessing and uses the locally installed build toolchains.

We evaluated both and yes, I would favour of having a mix of both projects, basically having a smart scheduler (icecc) with the ability to do remote processing (distcc) and using as well the locally installed toolchain (distcc)

**Comment by thedrow** [comment]
I think we should consider to use https://pypi.python.org/pypi/chardet to detect the encoding of the file. I'm aware it is another dependency to add but it will be much more reliable.

**Comment by shawnl** [comment]
Should be fixed with 23d362910b1a6b2dca8.
